src/closure.rs

```diff
@@ -53,23 +59,21 @@ impl Closure {
         }
 
         unsafe extern "C" fn finalize<F>(notify_data: *mut c_void, _closure: *mut gobject_ffi::GClosure)
-            where F: Fn(&[Value]) -> Option<Value> + Send + Sync + 'static
+            where F: Fn(&[Value]) -> Option<Value> + 'static
         {
             let _callback: Box<F> = Box::from_raw(notify_data as *mut _);
             // callback is dropped here.
         }
 
-        unsafe {
-            let size = 4 + 4 + 3 * mem::size_of::<*mut c_void>() as u32;
-            let closure = gobject_ffi::g_closure_new_simple(size, ptr::null_mut());
-            assert_ne!(closure, ptr::null_mut());
-            let callback = Box::new(callback);
-            let ptr: *mut F = Box::into_raw(callback);
-            let ptr: *mut c_void = ptr as *mut _;
-            gobject_ffi::g_closure_set_meta_marshal(closure, ptr, Some(marshal::<F>));
-            gobject_ffi::g_closure_add_finalize_notifier(closure, ptr, Some(finalize::<F>));
-            from_glib_none(closure)
-        }
+        let size = 4 + 4 + 3 * mem::size_of::<*mut c_void>() as u32;
```

Comment:
Could you comment what elements of this multiplication are please?

Reply:
Sure, but tomorrow :) Note that all your comments are already for the existing code. This code only lost a level of indentation

Reply:
Done, and also improved the calculation a bit